packages/clerk-js/src/core/resources/SignIn.ts

```diff
@@ -92,12 +93,11 @@ export class SignIn extends BaseResource implements SignInResource {
       case 'web3_metamask_signature':
         config = { web3WalletId: factor.web3WalletId } as Web3SignatureConfig;
         break;
-      case 'reset_password_code':
-        if (factor.emailAddressId) {
-          config = { emailAddressId: factor?.emailAddressId } as ResetPasswordCodeFactorConfig;
-        } else if (factor.phoneNumberId) {
-          config = { phoneNumberId: factor?.phoneNumberId } as ResetPasswordCodeFactorConfig;
-        }
+      case 'reset_password_phone_code':
+        config = { phoneNumberId: factor.phoneNumberId } as ResetPasswordPhoneCodeFactorConfig;
+        break;
+      case 'reset_password_email_code':
+        config = { emailAddressId: factor?.emailAddressId } as ResetPasswordEmailCodeFactorConfig;
```

Comment:
❓ Don't we need optional chaining in all config assignments, i.e. `factor?.phoneNumberId`?